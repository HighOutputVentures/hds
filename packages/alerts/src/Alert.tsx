import { ArrowForwardIcon, InfoOutlineIcon } from '@chakra-ui/icons';
import { CloseButton, Flex, HStack, Icon, Text } from '@chakra-ui/react';
// @ts-ignore
import { CheckCircleGreenIcon, WarningIcon } from '@highoutput/hds-icons';
import React from 'react';

export interface AlertsProps {
  status: 'info' | 'warning' | 'success' | 'error' | 'default';
  title: string;
  supportingDetail: string;
  alertLabel: {
    label1: string;
    label2?: string;
  };
  alertLinks: {
    link1: string | VoidFunction;
    link2?: string | VoidFunction;
  };
  isOpen?: boolean;
  onClose?: () => void;
}
const Alert = (props: AlertsProps) => {
  const {
    status,
    supportingDetail,
    title,
    isOpen,
    onClose,
    alertLabel,
    alertLinks,
  } = props;

  const AlertIcon =
    status === 'success'
      ? CheckCircleGreenIcon
      : status === 'warning'
      ? WarningIcon
      : InfoOutlineIcon;
  const BGColor =
    status === 'success'
      ? 'interface.success.500'
      : status === 'warning'
      ? 'interface.warning.500'
      : status === 'error'
      ? 'interface.error.500'
      : status === 'info'
      ? '#EDE8FC'
      : '#FCFCFD';

  const AlertTitleColor =
    status === 'success'
      ? 'interface.success.900'
      : status === 'warning'
      ? 'interface.warning.900'
      : status === 'error'
      ? 'interface.error.900 '
      : status === 'info'
      ? '#4A3880'
      : 'neutrals.900';
  const AlertDescColor =
    status === 'success'
      ? 'interface.success.800'
      : status === 'warning'
      ? 'interface.warning.800'
      : status === 'error'
      ? 'interface.error.800 '
      : status === 'info'
      ? '#5F48A3'
      : 'neutrals.800';
  const CloseBtnColor =
    status === 'success'
      ? 'interface.success.700'
      : status === 'warning'
      ? 'interface.warning.700'
      : status === 'error'
      ? 'interface.error.700 '
      : status === 'info'
      ? '#8A68EF'
      : 'neutrals.700';
  const BoxBorderColor =
    status === 'success'
      ? 'interface.success.600'
      : status === 'warning'
      ? 'interface.warning.600'
      : status === 'error'
      ? 'interface.error.600 '
      : status === 'info'
      ? '#C3B8E6'
      : 'neutrals.600';
  if (!isOpen) return null;
  return (
    <>
      <Flex
        padding={'16px'}
        align="start"
        bg={BGColor}
        borderWidth="1px"
        borderColor={`${BoxBorderColor}`}
        direction={['column', 'row']}
        maxW="1216px"
        position={'relative'}
        gap={'13.67px'}
        borderRadius={'8px'}
        data-testid="alert-box"
      >
        <Icon
          as={AlertIcon}
          position="relative"
          top={'3px'}
          color={AlertTitleColor}
        />
        <Flex direction="column">
          <Text
            color={AlertTitleColor}
            fontSize={'14px'}
            mb="4px"
            lineHeight="20px"
          >
            {title}
          </Text>
          <Text
            color={AlertDescColor}
            fontSize={'14px'}
            mb="12px"
            lineHeight="20px"
          >
            {supportingDetail}
          </Text>
          <HStack spacing="12px">
            <Text
              as="a"
              color={AlertDescColor}
              fontSize={'18px'}
              lineHeight="24px"
              fontWeight={500}
              cursor="pointer"
              href={`${alertLinks.link1}`}
              target={'_blank'}
            >
              {alertLabel.label1}
            </Text>
            <Text
              as="a"
              href={`${alertLinks.link2}`}
              target={'_blank'}
              cursor="pointer"
            >
              <Text
                as="span"
                color={status === 'default' ? '#8A68EF' : AlertTitleColor}
                fontSize={'18px'}
                lineHeight="24px"
                fontWeight={500}
                mr={'12px'}
              >
                {alertLabel.label2}
              </Text>

              <Icon
                as={ArrowForwardIcon}
                color={status === 'default' ? '#8A68EF' : AlertTitleColor}
                width={'20px'}
                height={'20px'}
              />
            </Text>
          </HStack>
        </Flex>
        <CloseButton
          position="absolute"
          right={'5px'}
          top={'5px'}
          color={CloseBtnColor}
          _hover={{ background: 'none' }}
          onClick={onClose}
        />
      </Flex>
    </>
  );
};

export default Alert;
