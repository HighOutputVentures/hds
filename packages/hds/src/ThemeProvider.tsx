import { ChakraProvider, ChakraProviderProps, extendTheme } from '@chakra-ui/react';
import * as React from 'react';
import Fonts from './Fonts';
import hdsTheme from './theme';

export interface ThemeProviderProps extends ChakraProviderProps {}

const ThemeProvider: React.FC<ThemeProviderProps> = ({ children, theme = {}, ...props }) => {
  return (
    <ChakraProvider theme={extendTheme(hdsTheme, theme)} {...props}>
      <Fonts />

      {children}
    </ChakraProvider>
  );
};

export default ThemeProvider;
