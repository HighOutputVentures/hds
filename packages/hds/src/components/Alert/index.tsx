export {
  Alert,
  AlertDescription,
  AlertDescriptionProps,
  AlertDialog,
  AlertDialogBody,
  AlertDialogCloseButton,
  AlertDialogContent,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogOverlay,
  AlertDialogProps,
  AlertIcon,
  AlertIconProps,
  AlertProps,
  AlertTitle,
  AlertTitleProps,
} from '@chakra-ui/react';
