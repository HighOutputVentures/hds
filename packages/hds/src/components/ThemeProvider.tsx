import { ChakraProvider, ChakraProviderProps } from '@chakra-ui/react';
import React, { FC } from 'react';

import { extendTheme } from '../utils/theme.utils';
import Fonts from './Fonts';

export interface ThemeProviderProps extends ChakraProviderProps {}

const ThemeProvider: FC<ThemeProviderProps> = ({ children, ...props }) => {
  return (
    <ChakraProvider {...props} theme={extendTheme(props.theme)}>
      <Fonts />
      {children}
    </ChakraProvider>
  );
};

export default ThemeProvider;
