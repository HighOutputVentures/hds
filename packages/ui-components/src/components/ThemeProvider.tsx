import { ChakraProvider, ChakraProviderProps } from '@chakra-ui/react';
import React, { FC } from 'react';

import '../Fonts/fonts.css';
import { extendTheme } from '../utils/theme.utils';

export interface ThemeProviderProps extends ChakraProviderProps {}

const ThemeProvider: FC<ThemeProviderProps> = ({ children, ...props }) => {
  return (
    <ChakraProvider {...props} theme={extendTheme(props.theme)}>
      {children}
    </ChakraProvider>
  );
};

export default ThemeProvider;
