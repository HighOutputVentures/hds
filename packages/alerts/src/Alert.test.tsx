import ThemeProvider from '@highoutput/hds/src/components/ThemeProvider';
import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Alert from './Alert';

function Component() {
  return (
    <Alert
      alertLabel={{ label1: 'Learn More', label2: 'View Changes' }}
      alertLinks={{ link1: '#', link2: '#' }}
      status="default"
      supportingDetail="Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid pariatur, ipsum similique veniam."
      title="Weâ€™ve just released a new feature"
      isOpen={true}
    />
  );
}

describe('Alert Component', () => {
  beforeEach(() => {
    render(
      <ThemeProvider>
        <Component />
      </ThemeProvider>
    );
  });

  it('Should render', async () => {
    const AlertBox = await screen.findAllByTestId('alert-box');
    expect(AlertBox).toHaveLength(1);
  });

  describe('Snapshot', () => {
    it('Should match snapshot', () => {
      const component = renderer.create(<Component />);

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      component.unmount();
    });
  });
});
