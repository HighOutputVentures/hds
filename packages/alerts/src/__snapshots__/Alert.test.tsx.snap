// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component Snapshot Should match snapshot 1`] = `
<div
  className="css-1ac0ctp"
  data-testid="alert-box"
>
  <svg
    className="chakra-icon chakra-icon css-hnz1ry"
    focusable={false}
    viewBox="0 0 24 24"
  >
    <g
      fill="currentColor"
      stroke="currentColor"
      strokeLinecap="square"
      strokeWidth="2"
    >
      <circle
        cx="12"
        cy="12"
        fill="none"
        r="11"
        stroke="currentColor"
      />
      <line
        fill="none"
        x1="11.959"
        x2="11.959"
        y1="11"
        y2="17"
      />
      <circle
        cx="11.959"
        cy="7"
        r="1"
        stroke="none"
      />
    </g>
  </svg>
  <div
    className="css-ohugnc"
  >
    <p
      className="chakra-text css-88nnmw"
    >
      Weâ€™ve just released a new feature
    </p>
    <p
      className="chakra-text css-1g8kskj"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid pariatur, ipsum similique veniam.
    </p>
    <div
      className="chakra-stack css-he6z7n"
    >
      <a
        className="chakra-text css-17o29vu"
        href="#"
        target="_blank"
      >
        Learn More
      </a>
      <a
        className="chakra-text css-4g6ai3"
        href="#"
        target="_blank"
      >
        <span
          className="chakra-text css-1ljeq08"
        >
          View Changes
        </span>
        <svg
          className="chakra-icon chakra-icon css-14adh8d"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    className="css-1t9pz9x"
  />
  <button
    aria-label="Close"
    className="css-1vcoxy9"
    type="button"
  >
    <svg
      aria-hidden={true}
      className="chakra-icon css-onkibi"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;
